include/oneapi/dpl/pstl/histogram_impl.h

```diff
@@ -77,6 +79,54 @@ histogram(_ExecutionPolicy&& exec, _RandomAccessIterator1 first, _RandomAccessIt
     return histogram_first + num_bins;
 }
 
+// Support extension API to cover existing API (previous to specification) with the following two overloads
+
+// This overload is provided to support an extension to the oneDPL specification to support the original implementation
+// of the histogram API, where the boundary type _ValueType could differ from the value type of the input iterator,
+// and required `<`, `<=`, `+`, `-`, and `/` to be defined between _ValueType and
+// std::iterator_traits<_RandomAccessIterator1>::value_type rather than enforcing they were the same type or convertible
+template <typename _ExecutionPolicy, typename _RandomAccessIterator1, typename _Size, typename _RandomAccessIterator2,
+          typename _ValueType>
+std::enable_if_t<
+    oneapi::dpl::execution::is_execution_policy_v<std::decay_t<_ExecutionPolicy>> &&
+        !std::is_convertible_v<_ValueType, typename std::iterator_traits<_RandomAccessIterator1>::value_type>,
+    _RandomAccessIterator2>
+histogram(_ExecutionPolicy&& exec, _RandomAccessIterator1 first, _RandomAccessIterator1 last, _Size num_bins,
+          _ValueType first_bin_min_val, _ValueType last_bin_max_val, _RandomAccessIterator2 histogram_first)
+{
+    const auto __dispatch_tag = oneapi::dpl::__internal::__select_backend(exec, first, histogram_first);
+
+    oneapi::dpl::__internal::__pattern_histogram(
+        __dispatch_tag, ::std::forward<_ExecutionPolicy>(exec), first, last, num_bins,
+        oneapi::dpl::__internal::__evenly_divided_binhash<_ValueType>(first_bin_min_val, last_bin_max_val, num_bins),
+        histogram_first);
+    return histogram_first + num_bins;
+}
+
+// This overload is provided to support an extension to the oneDPL specification to support the original implementation
+// of the histogram API, where if users explicitly-specify all template arguments, their arguments for bin boundary min
+// and max are convertible to the specified _ValueType, and the _ValueType is convertible to the value type of the
+// input iterator.
```

Comment:
It makes sense to mention that `_ValueType` is not deducible and so this overload can only be chosen if it's explicitly specified.

Reply:
Fixed.